packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,11 +72,244 @@ describe('Sayt Component', () => {
+      expect(showSayt).to.be.calledOnce;
+    });
+
+    it('should not call showSayt() when event specifies wrong searchbar ID', () => {
+      const showSayt = stub(sayt, 'showSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(false);
+      const event = {};
+
+      sayt.showCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(showSayt).to.not.be.called;
+    });
+  });
+
+  describe('hideCorrectSayt()', () => {
+    it('should call hideSayt() when event specifies correct searchbar ID ', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(true);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.be.calledOnce;
+    });
+
+    it('should not call hideSayt() when event specifies wrong searchbar ID', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(false);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('isCorrectSayt()', () => {
+    it('should return true if event provides the correct searchbar ID', () => {
+      const searchbar = sayt.searchbar = 'some-searchbar-id';
+      const event = { detail: { searchbar } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return true if event does not provide a searchbar ID', () => {
+      const searchbar = sayt.searchbar = 'some-searchbar-id';
+      const event = { detail: {} };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false if event provides the wrong searchbar ID', () => {
+      sayt.searchbar = 'correct-searchbar-id';
+      const event = { detail: { searchbar: 'wrong-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true if event specifies a searchbar but SAYT has none specified', () => {
+      sayt.searchbar = undefined;
+      const event = { detail: { searchbar: 'some-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should not error if passed an event without a detail attribute', () => {
+      const event = {};
+      const callback = () => sayt.isCorrectSayt(event);
+
+      expect(callback).to.not.throw();
+    });
+  });
+
   describe('render()', () => {
     it('should return an instance of TemplateResult', () => {
       const result = sayt.render();
 
       expect(result).to.be.an.instanceof(TemplateResult);
     });
   });
+
+  describe('processClick()', () => {
+    let node: any = 'some-node';
```

Comment:
If `target` is a `Node`, then `node` at minimum should be an object value so that the basic type is the same. If this is an object, then you'll have to set it to a new object every time in `beforeEach`.

Reply:
Why does this matter? According to the tests, it doesn't matter what node is. We are just asserting that it is passed along.

Reply:
There's a difference between objects and primitives in that the former is passed by reference, whereas the latter is passed by value. If you have a primitive where an object is expected, then you lose the ability to convincingly test whether it's the exact value or if it just has similar contents because both is the same for a primitive. In other words, you lose information by using a primitive. Doing this _may_ cause us to miss a bug that would otherwise have been caught by using an object.

Does `node` need to be a full `Node` object? Of course not, but I think it should still be an object, even if it's empty.

Furthermore, to paraphrase Uncle Bob, unit tests can be seen as a type of code sample, one that shows all the different ways that the class can be called. The samples are so correct and in-sync with the "production" code that you can run them. Having a string here signals to me that you can call these functions with strings, which is very much _not_ something you can do with these functions.

Reply:
Yes, the pointer argument makes sense. For the record, I disagree with Uncle Bob - unit tests are code samples only of the unit being tested. Integration tests would be an easier choice for his argument.